src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -16,7 +17,7 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<T> getAll(Class<? extends T> type);
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method. This is a good practice to simplify the method logic and enhance code readability.